packages/next/src/server/dev/hot-reloader.ts

```diff
@@ -413,6 +474,32 @@ export default class HotReloader {
       )
   }
 
+  private async getVersionInfo(span: Span) {
+    const versionInfoSpan = span.traceChild('get-version-info')
+    return versionInfoSpan.traceAsyncFn<VersionInfo>(async () => {
+      let installed = '0.0.0'
+      try {
+        installed = require('next/package.json').version
+
+        const res = await fetch(
+          'https://registry.npmjs.org/-/package/next/dist-tags'
```

Comment:
I thought we decided to use a cached endpoint with `stale-whlie-revalidate` to get the latest version?

If we're going to continue with fetching from `npm` we should honor the registry config similar to when downloading the wasm swc binary x-ref: https://github.com/vercel/next.js/blob/c6fe3f32240a92093a290847a3ae6023b01e3d04/packages/next/src/lib/download-wasm-swc.ts#L99

Reply:
We have to be careful with invoking `npm` cli because users who enable corepack for a project not using npm will error.

Reply:
We can make this a shared util and just use the active package manager which we have detection for already thankfully all of the CLIs have `config get registry` available.

Reply:
Good point! Created a helper for it and updated references.